src/controller/eme-controller.ts

```diff
@@ -1402,8 +1403,11 @@ class EMEController extends Logger implements ComponentAPI {
       if (index > -1) {
         this.mediaKeySessions.splice(index, 1);
       }
-      return mediaKeysSession
-        .remove()
+      const { drmSystemOptions } = this.config;
+      const removePromise = isPersistentSessionType(drmSystemOptions)
+        ? mediaKeysSession.remove()
```

Comment:
If this needs a timeout it could look like:

```suggestion
        ? new Promise((resolve, reject) => {
            self.setTimeout(
              () => reject(new Error(`MediaKeySession.remove() timeout`)),
              8000,
            );
            mediaKeysSession.remove().then(resolve);
          })
```